libraries/AP_HAL_Linux/RCInput_RPI.cpp

```diff
@@ -450,11 +463,17 @@ void RCInput_RPI::init()
 
     //Reading first sample
     curr_tick = *((uint64_t*) circle_buffer->get_page(circle_buffer->_virt_pages, curr_pointer));
-    prev_tick = curr_tick;
     curr_pointer += 8;
-    curr_signal = *((uint8_t*) circle_buffer->get_page(circle_buffer->_virt_pages, curr_pointer)) & 0x10 ? 1 : 0;
-    last_signal = curr_signal;
-    curr_pointer++;
+    signal_states = *((uint64_t*) circle_buffer->get_page(circle_buffer->_virt_pages, curr_pointer));
+    for (uint32_t i = 0; i < RCIN_RPI_CHN_NUM; ++i) {
+        rc_channels[i].prev_tick = curr_tick;
+        rc_channels[i].curr_signal = (signal_states & (1 << RcChnGpioTbl[i])) ? RCIN_RPI_SIG_HIGH
+                                                                              : RCIN_RPI_SIG_LOW;
+        rc_channels[i].last_signal = rc_channels[i].curr_signal;
+    }
+    curr_pointer += 8;
+
+    set_num_channels(RCIN_RPI_CHN_NUM);
```

Comment:
I think this call here is not correct. At least for PPMSUM: you would be setting the number of channels to 1. For ppm it will end up overridden later by ` _process_ppmsum_pulse()` so it's harmless, but not so correct IMO.